Extension/src/LanguageServer/lmTool.ts

```diff
@@ -44,6 +48,129 @@ const knownValues: { [Property in keyof ChatContextResult]?: { [id: string]: str
     }
 };
 
+function formatChatContext(context: ChatContextResult | ProjectContextResult): void {
+    type KnownKeys = 'language' | 'standardVersion' | 'compiler' | 'targetPlatform';
+    for (const key in knownValues) {
+        const knownKey = key as KnownKeys;
+        if (knownValues[knownKey] && context[knownKey]) {
+            // Clear the value if it's not in the known values.
+            context[knownKey] = knownValues[knownKey][context[knownKey]] || "";
+        }
+    }
+}
+
+export interface ProjectContext {
+    language: string;
+    standardVersion: string;
+    compiler: string;
+    targetPlatform: string;
+    targetArchitecture: string;
+    compilerArguments: string[];
+}
+
+export function getCompilerArgumentFilterMap(compiler: string, context: { flags: Record<string, unknown> }): { [key: string]: string } | undefined {
+    // The copilotcppXXXCompilerArgumentFilters are maps.
+    // The keys are regex strings and the values, if not empty, are the prompt text to use when no arguments are found.
+    let filterMap: { [key: string]: string } | undefined;
+    try {
+        switch (compiler) {
+            case MSVC:
+                filterMap = context.flags.copilotcppMsvcCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppMsvcCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+            case Clang:
+                filterMap = context.flags.copilotcppClangCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppClangCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+            case GCC:
+                filterMap = context.flags.copilotcppGccCompilerArgumentFilter ? JSON.parse(context.flags.copilotcppGccCompilerArgumentFilter as string ?? '{}') : undefined;
+                break;
+        }
+    }
+    catch {
+        // Intentionally swallow any exception.
+    }
+    return filterMap;
+}
+
+function filterComplierArguments(compiler: string, compilerArguments: string[], context: { flags: Record<string, unknown> }): string[] {

```

Comment:
filterComplierArguments->filterCompilerArguments

However, as I mentioned elsewhere, I'm not sure why the filtering is done 2 times or if this method is actually needed or not.

Reply:
Fixed the code to only filter once, thanks.